engine.py

```diff
@@ -1377,6 +1377,13 @@ def show_dialog(self, title, bundle, widget_class, *args, **kwargs):
         # make sure the window raised so it doesn't
         # appear behind the main After Effects window
         self.logger.debug("Showing dialog: %s" % (title,))
+
+        # Adding this window flag will ensure that our dialogs won't be
+        # tabbed together into a single window on OSX.
+        if sys.platform == "darwin":
+            from sgtk.platform.qt import QtCore
+            dialog.setWindowFlags(dialog.windowFlags() | QtCore.Qt.CustomizeWindowHint)
```

Comment:
Should this fix be in tk-core instead? I think this problem happens in Photoshop as well, right? Maybe in the TankQDialog?